client/modules/IDE/components/SketchList.jsx

```diff
@@ -179,6 +208,56 @@ class SketchListRowBase extends React.Component {
             ref={this.renameInput}
           />
         )}
+
+        {this.state.visibleDialogOpen && (
+          <Overlay
+            title={title}
+            closeOverlay={() => this.setState({ visibleDialogOpen: false })}
+          >
+            {/* TODO: these <li> should come from translate.json */}
+            <div className="sketch-visibility">
```

Comment:
I'm wondering if we should move the `VisibilityDialog` to its own component.  I'm thinking that we probably want to use this for confirmation when clicking the lock icon on the sketch page.  Otherwise it might be too easy to make something public by mistake?

Reply:
Yes, you're right. I've also considered that. However, since this is part of a complete private sketch feature, I believe there will be quite a discussion about the UI. Later on, I will move it to its own component once the UI is finalized. I spent more than an hour trying to find two symmetric good icons and align it perfectly inside the row of `SketchList.jsx` (yet it seems like I've left some missing pieces). I didn't realize how messy it would be to manage SVG directly. Perhaps we might want to use React-Icons?

Reply:
Additionally, I don't know if you have noticed this, but the `<Overlay />` component overwrites any toast message.

Reply:
> Additionally, I don't know if you have noticed this, but the `<Overlay />` component overwrites any toast message.

Hmm.  I have not noticed it.  That's concerning.  Feel free to open a new issue explaining in more detail.